transactions/repository.go

```diff
@@ -0,0 +1,78 @@
+package transactions
+
+import (
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/storage"
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/satori/go.uuid"
+)
+
+const (
+	// ErrTransactionMissing error when transaction doesn't exist in Repository.
+	ErrTransactionMissing = errors.Error("transaction doesn't exist")
+
+	// ErrKeyConstructionFailed error when the key construction failed.
+	ErrKeyConstructionFailed = errors.Error("failed to construct transaction key")
+)
+
+// Repository can be implemented by a type that handles storage for transactions.
+type Repository interface {
+	Get(identity common.Address, id uuid.UUID) (*Transaction, error)
+	Save(transaction *Transaction) error
+}
+
+// txRepository implements Repository.
+type txRepository struct {
+	repo storage.Repository
+}
+
+// NewRepository registers the the transaction model and returns the an implementation
+// of the Repository.
+func NewRepository(repo storage.Repository) Repository {
+	repo.Register(&Transaction{})
+	return &txRepository{repo: repo}
+}
+
+// getKey appends identity with id.
+// With identity coming at first, we can even fetch transactions belonging to specific identity through prefix.
+func getKey(identity common.Address, id uuid.UUID) ([]byte, error) {
```

Comment:
I know we would use Ethereum addresses for identities everywhere soon, but I prefer to make the change 
from
`type CentID [len]byte`
to
`type CentID common.Address` in `identities` package
than use `common.Address` in these places, it's a helpful abstraction and this kind of change is exactly the reason why we have it. Let's just use `CentID` type here for all function signatures.

Reply:
We are purposefully not calling it CentID but rather identity in our epics so that it is not specific. Maybe @lucasvo or @manuelpolzhofer can explain more?

Reply:
We decided to name it identityAddress or identity instead.

I would suggest: DID (Decentralized Identifier)

```golang
type DID common.Address
```
I think keeping a specific type a good idea.

Reply:
lets do the rename later, but lets use CentID for now.